src/sentry/utils/kvstore/bigtable.py

```diff
@@ -0,0 +1,287 @@
+import enum
+import logging
+import struct
+import zlib
+from datetime import timedelta
+from threading import Lock
+from typing import Any, Callable, Iterator, Mapping, NamedTuple, Optional, Sequence, Tuple, cast
+
+import zstandard
+from django.utils import timezone
+from google.api_core import exceptions, retry
+from google.cloud import bigtable
+from google.cloud.bigtable.row_data import PartialRowData
+from google.cloud.bigtable.row_set import RowSet
+from google.cloud.bigtable.table import Table
+from sentry.utils.kvstore.abstract import KVStorage
+
+
+logger = logging.getLogger(__name__)
+
+
+class CompressionStrategy(NamedTuple):
+    compress: Callable[[bytes], bytes]
+    decompress: Callable[[bytes], bytes]
+
+
+class BigtableError(Exception):
+    pass
+
+
+class BigtableKVStorage(KVStorage[str, bytes]):
+    max_size = 1024 * 1024 * 10
+    column_family = "x"
+
+    ttl_column = b"t"
+    ttl_struct = struct.Struct("<I")
+    flags_column = b"f"
+    flags_struct = struct.Struct("B")
+    data_column = b"0"
+
+    class Flags(enum.IntFlag):
```

Comment:
This has to be `IntFlag` and not `Flag` so that it can be instantiated with bits that aren't covered by an existing option:

```
>>> from enum import Flag, IntFlag
>>> 
>>> class Test(Flag):
...   a = 1
... 
>>> class IntTest(IntFlag):
...   a = 2
...
>>> Flag(3)
ValueError: 3 is not a valid Flag

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ted/.pyenv/versions/3.8.3/lib/python3.8/enum.py", line 304, in __call__
    return cls.__new__(cls, value)
  File "/Users/ted/.pyenv/versions/3.8.3/lib/python3.8/enum.py", line 595, in __new__
    raise exc
  File "/Users/ted/.pyenv/versions/3.8.3/lib/python3.8/enum.py", line 579, in __new__
    result = cls._missing_(value)
  File "/Users/ted/.pyenv/versions/3.8.3/lib/python3.8/enum.py", line 699, in _missing_
    possible_member = cls._create_pseudo_member_(value)
  File "/Users/ted/.pyenv/versions/3.8.3/lib/python3.8/enum.py", line 714, in _create_pseudo_member_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 3 is not a valid Flag
>>> IntFlag(3)
<IntFlag.2|1: 3>
```